### PR TITLE
BUG: Fix for broken alma downloader

### LINKED ISSUE #2489: ALMA data download broken
The ALMA data downloader is broken because of an upstream change:

```python
from astroquery.alma import Alma
alma = Alma()
uid = 'uid://A001/X12a3/Xe9'
data_info = alma.get_data_info(uid, expand_tarfiles=True)
alma.retrieve_data_from_uid([uid])
```

This fails on the second entry in the list because there is now a blank URL in the `access_url` column:
```
>>>  data_info
<Table length=9>
         ID                                                   access_url                                                                  service_def                         error_message   semantics                           description                           content_type   content_length readable
                                                                                                                                                                                                                                                                                            byte
       object                                                   object                                                                       object                               object        object                               object                                object          int64        bool
-------------------- -------------------------------------------------------------------------------------------- ----------------------------------------------------------- ------------- -------------- --------------------------------------------------------- ----------------- -------------- --------
uid://A001/X12a3/Xe9               https://almascience.nrao.edu/dataPortal/member.uid___A001_X12a3_Xe9.README.txt                                                                           #documentation Download documentation for ./uid://A001/X12a3/Xe9|README.        text/plain           3523     True
uid://A001/X12a3/Xe9   https://almascience.nrao.edu/dataPortal/2017.1.01185.S_uid___A001_X12a3_Xe9_001_of_001.tar                                                                                    #this           Download dataset of type: null, and class: N/A. application/x-tar      556278784     True
uid://A001/X12a3/Xe9                                                                                              DataLink.2017.1.01185.S_uid___A001_X12a3_Xe9_001_of_001.tar                        #this                                                                    text/xml             --       --
uid://A001/X12a3/Xe9    https://almascience.nrao.edu/dataPortal/2017.1.01185.S_uid___A001_X12a3_Xe9_auxiliary.tar                                                                               #auxiliary           Download dataset of type: null, and class: N/A. application/x-tar      135479296     True
uid://A001/X12a3/Xe9                                                                                               DataLink.2017.1.01185.S_uid___A001_X12a3_Xe9_auxiliary.tar                   #auxiliary                                                                    text/xml             --       --
uid://A001/X12a3/Xe9 https://almascience.nrao.edu/dataPortal/2017.1.01185.S_uid___A002_Xd28a9e_X71b8.asdm.sdm.tar                                                                              #progenitor               Download dataset of type: , and class: N/A. application/x-tar     2343569408     True
uid://A001/X12a3/Xe9 https://almascience.nrao.edu/dataPortal/2017.1.01185.S_uid___A002_Xd28a9e_X7b4d.asdm.sdm.tar                                                                              #progenitor               Download dataset of type: , and class: N/A. application/x-tar     2343861248     True
uid://A001/X12a3/Xe9 https://almascience.nrao.edu/dataPortal/2017.1.01185.S_uid___A002_Xd29c1f_X1f74.asdm.sdm.tar                                                                              #progenitor               Download dataset of type: , and class: N/A. application/x-tar     2616798208     True
uid://A001/X12a3/Xe9  https://almascience.nrao.edu/dataPortal/2017.1.01185.S_uid___A002_Xd29c1f_X5cf.asdm.sdm.tar                                                                              #progenitor               Download dataset of type: , and class: N/A. application/x-tar     2680167424     True
```

The error is:
```python
Traceback (most recent call last):
  File "<ipython-input-13-3e41ffe75afd>", line 1, in <module>
    alma.retrieve_data_from_uid([uid])
  File "/home/adam/repos/astroquery/astroquery/alma/core.py", line 807, in retrieve_data_from_uid
    downloaded_files = self.download_files(file_urls)
  File "/home/adam/repos/astroquery/astroquery/alma/core.py", line 681, in download_files
    check_filename = self._request('HEAD', file_link, auth=auth)
  File "/home/adam/repos/astroquery/astroquery/query.py", line 315, in _request
    response = query.request(self._session,
  File "/home/adam/repos/astroquery/astroquery/query.py", line 69, in request
    return session.request(self.method, self.url, params=self.params,
  File "/home/adam/anaconda3/lib/python3.8/site-packages/requests/sessions.py", line 528, in request
    prep = self.prepare_request(req)
  File "/home/adam/anaconda3/lib/python3.8/site-packages/requests/sessions.py", line 456, in prepare_request
    p.prepare(
  File "/home/adam/anaconda3/lib/python3.8/site-packages/requests/models.py", line 316, in prepare
    self.prepare_url(url, params)
  File "/home/adam/anaconda3/lib/python3.8/site-packages/requests/models.py", line 390, in prepare_url
    raise MissingSchema(error)
MissingSchema: Invalid URL '': No schema supplied. Perhaps you meant http://?
```

@andamian I think this is a simple fix in astroquery, I'll propose the fix, but I hope you understand something about the upstream change.

### PR DESCRIPTION
**Pull request by keflavich**
This should fix #2489.  The failure is already covered by tests.

### CODE REVIEW

**Review by ceb8 — APPROVED** [review]
LGTM

**Comment by keflavich** [comment]
Thanks.  I'd like to have @andamian 's eyes on this before merging in case there is a reason to take a different approach (e.g., there are some other metadata we should be inspecting to determine what action to take in these blank-line cases)

**Comment by bsipocz** [comment]
@keflavich - I don't see any of the tests to fail even without this PR, so they'll need to be adjusted to make sure they catch the pre-PR regression.

Also, I haven't looked into the details, but recall that https://github.com/astropy/astroquery/pull/2438 might be in play here?

**Review by bsipocz — CHANGES_REQUESTED** [review]
Tests should be added/refactored for this. Also it needs a changelog.

**Comment by at88mph** [comment]
This is already addressed in #2438 .

**Comment by keflavich** [comment]
OK, then I'll defer to that PR, but it's not at all obvious to me how this is addressed there.   The underlying infrastructure is opaque to me.

The test that needs to fail is indeed marked `xfail` because it's awaiting #2438.

So I guess the new datalink service was deployed in the last ~48 hours?  I have a daily cron job that started failing today, but was fine yeserday.

**Comment by bsipocz** [comment]
I'll try to get back to that PR this afternoon, as I recall my reservations were all about the modifications to the generic tooling, and nothing about the specifics in the alma module.

**Comment by bsipocz** [comment]
Closing this in favour of #2493

**Comment by bsipocz** [comment]
Are those datalinks copies of the same products that are already been listed with non-blank access_url? 

For merging, this will need the review and approval of someone knowing the details of the ALMA API, @at88mph, or @andamian.

**Comment by keflavich** [comment]
> Are those datalinks copies of the same products that are already been listed with non-blank access_url?

afaict, yes, the blank lines correspond to the tarballs that are now excluded from the download b/c they've been expanded.

There is likely to be a more rational way to handle this, but this is the bugfix I need know because the currently deployed version is broken.  So, if other users stumble across this before we resolve the issue, this PR is an acceptable _temporary_ workaround.

**Comment by keflavich** [comment]
@at88mph @andamian will either of you be able to review this PR this week? 

If not, I recommend we merge it and request post-merge review to see if there is a better solution, but we need this as a patch now.

File: astroquery/alma/tests/test_alma_remote.py
Comment (eerovaher) [inline astroquery/alma/tests/test_alma_remote.py:157]: Is there a good reason not to use the [`temp_path` fixture](https://docs.pytest.org/en/stable/how-to/tmp_path.html) from `pytest` itself?
```suggestion
    def test_retrieve_data(self, temp_path, alma):
        """
        Regression test for issue 2490 (the retrieval step will simply fail if
        given a blank line, so all we're doing is testing that it runs)
        """
        alma.cache_location = temp_path
```

Comment (keflavich) [reply astroquery/alma/tests/test_alma_remote.py:157]: not that I know of; I just copy-pasted another test

**Comment by at88mph** [comment]
Good catch @keflavich, thanks for this.

**Comment by bsipocz** [comment]
I take the comment of @at88mph from above as an approval, so am going ahead and merging this now. Thanks, @keflavich!

**Pull request merged**
